bundles/org.openhab.binding.guntamatic/src/main/java/org/openhab/binding/guntamatic/internal/GuntamaticHandler.java

```diff
@@ -0,0 +1,426 @@
+    private final HttpClient httpClient;
+    private GuntamaticChannelTypeProvider guntamaticChannelTypeProvider;
+    private HashMap<Integer, String> channels = new HashMap<Integer, String>();
+    private HashMap<Integer, String> types = new HashMap<Integer, String>();
+    private HashMap<Integer, String> units = new HashMap<Integer, String>();
+
+    public GuntamaticHandler(Thing thing, HttpClient httpClient,
+            GuntamaticChannelTypeProvider guntamaticChannelTypeProvider) {
+        super(thing);
+        this.httpClient = httpClient;
+        this.guntamaticChannelTypeProvider = guntamaticChannelTypeProvider;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (!(command instanceof RefreshType)) {
+            if (!config.key.isEmpty()) {
+                String param;
+                String channelID = channelUID.getId();
+                switch (channelID) {
+                    case CHANNEL_CONTROLBOILERAPPROVAL:
+                        param = getThing().getProperties().get(PARAMETER_BOILERAPPROVAL);
+                        break;
+                    case CHANNEL_CONTROLPROGRAM:
+                        param = getThing().getProperties().get(PARAMETER_PROGRAM);
+                        break;
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM0:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM1:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM2:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM3:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM4:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM5:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM6:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM7:
+                    case CHANNEL_CONTROLHEATCIRCPROGRAM8:
+                        param = getThing().getProperties().get(PARAMETER_HEATCIRCPROGRAM).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    case CHANNEL_CONTROLWWHEAT0:
+                    case CHANNEL_CONTROLWWHEAT1:
+                    case CHANNEL_CONTROLWWHEAT2:
+                        param = getThing().getProperties().get(PARAMETER_WWHEAT).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    case CHANNEL_CONTROLEXTRAWWHEAT0:
+                    case CHANNEL_CONTROLEXTRAWWHEAT1:
+                    case CHANNEL_CONTROLEXTRAWWHEAT2:
+                        param = getThing().getProperties().get(PARAMETER_EXTRAWWHEAT).replace("x",
+                                channelID.substring(channelID.length() - 1));
+                        break;
+                    default:
+                        return;
+                }
+                String response = sendGetRequest(PARSET_URL, "syn=" + param + "&value=" + command.toString());
```

Comment:
Considering how you defined sendGetRequest, you could rather call it like that:
` String response = sendGetRequest(PARSET_URL, "syn=" + param, "value=" + command.toString());`

Reply:
`String response = sendGetRequest(PARSET_URL, "syn=" + param, "value=" + command.toString());`